libnmstate/nm/sriov.py

```diff
@@ -162,42 +140,69 @@ def get_info(device):
-    connection.import_by_device(device)
-    if not connection.profile:
-        info[Ethernet.SRIOV_SUBTREE] = sriov_config
-        return info
-
-    sriov_setting = connection.profile.get_setting_by_name(
-        nmclient.NM.SETTING_SRIOV_SETTING_NAME
-    )
-
-    if sriov_setting:
-        vfs_config = _get_info_sriov_vfs_config(sriov_setting)
-        sriov_config[Ethernet.SRIOV.VFS_SUBTREE] = vfs_config
+    if sriov_config[Ethernet.SRIOV.TOTAL_VFS]:
+        sriov_config[
+            Ethernet.SRIOV.VFS_SUBTREE
+        ] = _get_sriov_vfs_configuration(ifname)
+    else:
+        sriov_config[Ethernet.SRIOV.VFS_SUBTREE] = []
 
-        info[Ethernet.SRIOV_SUBTREE] = sriov_config
+    info[Ethernet.SRIOV_SUBTREE] = sriov_config
 
     return info
 
 
-def _get_info_sriov_vfs_config(sriov_setting):
-    vfs_config = []
-    vfs_setting = sriov_setting.props.vfs
-    for vf in vfs_setting:
-        vf_config = {}
-        vf_config[Ethernet.SRIOV.VFS.ID] = vf.get_index()
-        for nm_attribute in SRIOV_NM_TO_NMSTATE_MAP.keys():
-            _get_nm_attribute(vf, vf_config, nm_attribute)
+def _get_sriov_vfs_configuration(ifname):
+    """
+    This is a workaround to get the VFs configuration from runtime.
+    Ref: https://bugzilla.redhat.com/1777520
+    """
+    proc = subprocess.run(
+        ("ip", "link", "show", ifname),
+        stdout=subprocess.PIPE,
+        encoding="utf-8",
+    )
+    iplink_output = proc.stdout
 
-        vfs_config.append(vf_config)
+    vfs = iplink_output.splitlines(False)[2:]
+    vfs_config = [vf_config for vf_config in _create_vf_config(vfs)]
 
     return vfs_config
 
 
-def _get_nm_attribute(vf_object, vf_config, nm_attribute):
-    nmstate_key, nm_variant = SRIOV_NM_TO_NMSTATE_MAP[nm_attribute]
-    if vf_object.get_attribute(nm_attribute) is not None:
-        vf_config[nmstate_key] = nm_variant(
-            vf_object.get_attribute(nm_attribute)
-        )
+def _create_vf_config(vfs):
+    mac_expr = re.compile(r"[a-fA-F0-9:]{17}|[a-fA-F0-9]{12}")
+    min_tx_expr = re.compile(r"min_tx_rate [0-9]*")
+    max_tx_expr = re.compile(r"max_tx_rate [0-9]*")
+    spoof_expr = re.compile(r"checking (on|off)")
+    trust_expr = re.compile(r"trust (on|off)")
+    vf_id = 0
+    for vf in vfs:
+        vf_config = {
+            Ethernet.SRIOV.VFS.ID: vf_id,
+            Ethernet.SRIOV.VFS.MAC_ADDRESS: mac_expr.search(vf).group(0),
+        }
+
+        spoof = spoof_expr.search(vf).group(0).split()[1]
+        if spoof == "on":
+            spoof = True
+        else:
+            spoof = False
+        vf_config[Ethernet.SRIOV.VFS.SPOOF_CHECK] = spoof
+        trust = trust_expr.search(vf).group(0).split()[1]
+        if trust == "on":
+            trust = True
+        else:
+            trust = False
+        vf_config[Ethernet.SRIOV.VFS.TRUST] = trust
+        min_tx = min_tx_expr.search(vf)
+        if min_tx:
+            min_tx = min_tx.group(0).split()[1]
```

Comment:
You can just change regex group like `max_tx_rate ([0-9]*)` instead of doing `split()`.

Since you do `split()[1]`, you are expecting there is a number after it, the regex should be `max_tx_rate ([0-9]+)`.

Reply:
Done and then using `.group(1)` in order to match it.